### ISSUE TITLE
0.13: [Bug]: ModuleTemplate failed to identify variable with default value pass in schema.json file

### ISSUE THREAD

**Issue by ishantagarwal-x213235** [garden-io/garden#4882]
### Current Behavior

Garden fails with the error:

```console
Failed resolving one or more modules:

name: Invalid template string (-Djib.from.image=${var.dockerRepo}…): Could not find key jdkVersion under inputs. Available keys: (none).
```

### Expected behavior

No errors, validation should pass.

### Reproducible example

Run `garden validate` with the configuration below.

**Garden configuration.**
```yaml
kind: Project
apiVersion: garden.io/v1
name: repro
environments:
  - name: default
providers:
  - name: local-kubernetes
    environments: [ default ]
variables:
  dockerRepo: "hub.docker.io"
  baseImageJava8: eclipse-temurin:8u345-b01-jre-ubuntu23
  baseImageJava11: distroless-os-images:debian11-java11.0.18

---
kind: Module
type: templated
template: k8s-container
name: name

---
kind: ModuleTemplate
name: k8s-container
inputsSchemaPath: schema.json

modules:
  - kind: Module
    type: container
    name: ${parent.name}  # Now possible as it doesn't conflict with the RenderTemplate type
    extraFlags:
      - "-Djib.from.image=${var.dockerRepo}/${inputs.jdkVersion == '11' ? var.baseImageJava11 : var.baseImageJava8}"
```

**`schema.json`**
```yaml
{
    "type": "object",
    "properties": {
      "excludeModules": {
        "type": "array",
        "items": {
            "type": "string"
        }
      },
      "jdkVersion": {
        "type": "string",
        "default": "8"
      }
    }
}
```

**Dummy `Dockerfile`**
```dockerfile
FROM busybox:1.36.0
ENTRYPOINT ["tail", "-f", "/dev/null"]
```

### Workaround

Use action-based template:

```yaml
---
kind: ModuleTemplate
name: k8s-container
inputsSchemaPath: schema.json

configs:
  - kind: Build
    type: container
    name: ${parent.name}  # Now possible as it doesn't conflict with the RenderTemplate type
    spec:
      extraFlags:
        - "-Djib.from.image=${var.dockerRepo}/${inputs.jdkVersion == '11' ? var.baseImageJava11 : var.baseImageJava8}"
```

**Comment by vvagaytsev** [comment]
@ishantagarwal-x213235 I've also updated the `ModuleTemplate` config. The `extraFlags` of the `Deploy` action must be in located in the `spec`:

```yaml
---
kind: ModuleTemplate
name: k8s-container
inputsSchemaPath: schema.json

configs:
  - kind: Build
    type: container
    name: ${parent.name}  # Now possible as it doesn't conflict with the RenderTemplate type
    spec:
      extraFlags:
        - "-Djib.from.image=${var.dockerRepo}/${inputs.jdkVersion == '11' ? var.baseImageJava11 : var.baseImageJava8}"
```

**Comment by vvagaytsev** [comment]
It looks like the failure comes from the [function `resolveModuleConfig`](https://github.com/garden-io/garden/blob/main/core/src/resolve-module.ts#L378-L380).

There we explicitly provide empty `inputs: {}` and `variables: {}`. Was that done intentionally @edvald @thsig? Can it be the cause of the failure here? cc @garden-io/core

**Comment by thsig** [comment]
The inputs are resolved above:

https://github.com/garden-io/garden/blob/main/core/src/resolve-module.ts#L368-L370

And the variables should already be taken care of by the call to `this.resolveVariables` here:

https://github.com/garden-io/garden/blob/main/core/src/resolve-module.ts#L365

I may be lacking a bit of context here, but it's not immediately obvious to me that the problem comes from here.

Maybe there's a problem in the way the module templates are being converted into config templates? I wonder if the inputs are being lost there somehow. If you're able to repro this, I'd recommend stepping through this with a debugger to verify that the inputs are being copied through correctly.

https://github.com/garden-io/garden/blob/main/core/src/garden.ts#L1347

Regardless of where the problem lies, we should use the opportunity to write a few more test cases around this flow.

**Comment by vvagaytsev** [comment]
@thsig right, I. think I saw some lost inputs in [function `scanAndAddConfigs`](https://github.com/garden-io/garden/blob/main/core/src/garden.ts#L1270). I'll take a closer look and compare transformation flows for `modules`- and `configs`-based templates.

**Comment by edvald** [comment]
Looks like we need more high-level tests around this. Would be great to get those in to start and the source of the problem will reveal itself from there.

**Comment by vvagaytsev** [comment]
The fix will be released in `0.13.10`.
